proposals/lip-0057.md

```diff
@@ -1940,12 +1994,16 @@ def afterTransactionsExecute(b: Block) -> None:
 
     # Update previousTimestamp substore.
     previousTimestamp = b.header.timestamp
+```
 
-    if isEndOfRound(height) == False:
-        return
+```python
+def updateSnapshotSubstore(height: uint32) -> None:
+    roundNumber = getRoundNumberFromHeight(height+1)
+    snapshotRound = roundNumber + 2
```

Comment:
Is there are specific reason why you don't do 
```
roundNumber = getRoundNumberFromHeight(height)
snapshotRound = roundNumber + 3
```
This would somehow be more canonical in how you define the snapshot store at the beginning. I would also try to avoid to use the generic name `roundNumber` for any round number, but be more specific, i.e., `currentRound`, `nextRound`.

Reply:
Replaced `roundNumber` by `nextRound`. The particular choice was made in alignment with the implementation and after discussion with Shu where we agreed that it's better to make clear that at last block of round we select validators for round +1 and the offset of rounds is 2. I agree there are many equivalent ways to define the same thing and I would just leave it as is now.